File: php/features.php
<?php
declare(strict_types=1);

// Nekosbest SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NekosbestFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NekosbestBaseFeature();
            case "test":
                return new NekosbestTestFeature();
            default:
                return new NekosbestBaseFeature();
        }
    }
}

File: py/features.py
# Nekosbest SDK feature factory

from feature.base_feature import NekosbestBaseFeature
from feature.test_feature import NekosbestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NekosbestBaseFeature(),
        "test": lambda: NekosbestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
